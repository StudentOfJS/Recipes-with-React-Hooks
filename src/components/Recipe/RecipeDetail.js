import React, { memo } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const RecipeDetail = memo(({ className, detail, style }) => (
  <div style={style} className={classNames('p2 bg-white rounded', className)}>
    <span>
      <h2 className="h2">{detail.name}</h2>
      <h4>{detail.category}</h4>
    </span>
    <img src={detail.image} alt={detail.name} className="rounded fit" />
    <h3>Ingredients</h3>
    <ul>
      {detail.ingredients.map((ingredient, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <li key={`ingredient-${i}`}>{ingredient}</li>
      ))}
    </ul>
    <h3>Steps</h3>
    <ol>
      {detail.steps.map((step, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <li key={`step-${i}`}>{step}</li>
      ))}
    </ol>
  </div>
));

RecipeDetail.propTypes = {
  className: PropTypes.string,
  detail: PropTypes.shape({
    name: PropTypes.string,
    image: PropTypes.string,
    category: PropTypes.string,
    calories: PropTypes.number,
    ingredients: PropTypes.array,
    steps: PropTypes.array,
  }),
  // eslint-disable-next-line react/forbid-prop-types
  style: PropTypes.object,
};

RecipeDetail.defaultProps = {
  className: 'ml4',
  detail: {
    name: '',
    image: '',
    category: '',
    calories: 0,
    ingredients: [],
    steps: [],
  },
  // eslint-disable-next-line react/forbid-prop-types
  style: { flex: 5 },
};

export default RecipeDetail;
