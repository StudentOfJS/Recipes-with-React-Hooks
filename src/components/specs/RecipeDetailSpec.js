import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RecipeDetail from '../Recipe/RecipeDetail';

const testRecipe = {
  detail: {
    name: 'test recipe',
    image: '',
    category: 'test category',
    calories: 1000000000,
    ingredients: ['ing1', 'ing2'],
    steps: ['step1', 'step2'],
  },
};

describe('<RecipeDetail />', () => {
  test('should render without props', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should correctly render a recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail {...testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should correctly render a className', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail className="py6" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
