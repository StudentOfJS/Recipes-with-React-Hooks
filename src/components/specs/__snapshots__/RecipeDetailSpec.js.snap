// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> should correctly render a className 1`] = `
<div
  className="p2 bg-white rounded py6"
  style={
    Object {
      "flex": 5,
    }
  }
>
  <span>
    <h2
      className="h2"
    >
      
    </h2>
    <h4>
      
    </h4>
  </span>
  <img
    alt=""
    className="rounded fit"
    src=""
  />
  <h3>
    Ingredients
  </h3>
  <ul />
  <h3>
    Steps
  </h3>
  <ol />
</div>
`;

exports[`<RecipeDetail /> should correctly render a recipe 1`] = `
<div
  className="p2 bg-white rounded ml4"
  style={
    Object {
      "flex": 5,
    }
  }
>
  <span>
    <h2
      className="h2"
    >
      test recipe
    </h2>
    <h4>
      test category
    </h4>
  </span>
  <img
    alt="test recipe"
    className="rounded fit"
    src=""
  />
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      ing1
    </li>
    <li>
      ing2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step1
    </li>
    <li>
      step2
    </li>
  </ol>
</div>
`;

exports[`<RecipeDetail /> should render without props 1`] = `
<div
  className="p2 bg-white rounded ml4"
  style={
    Object {
      "flex": 5,
    }
  }
>
  <span>
    <h2
      className="h2"
    >
      
    </h2>
    <h4>
      
    </h4>
  </span>
  <img
    alt=""
    className="rounded fit"
    src=""
  />
  <h3>
    Ingredients
  </h3>
  <ul />
  <h3>
    Steps
  </h3>
  <ol />
</div>
`;
